File: trade/src/main/java/com/ez08/trade/ui/view/InputCodeView.java
package com.ez08.trade.ui.view;

import android.content.Context;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.AttributeSet;
import android.widget.EditText;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.ez08.trade.R;
import com.ez08.trade.tools.ScreenUtil;

public class InputCodeView extends RelativeLayout {

    EditText editText;
    TextView textView;

    public InputCodeView(Context context) {
        super(context);
    }

    public InputCodeView(Context context, AttributeSet attrs) {
        super(context, attrs);
        inflate(context, R.layout.trade_view_input_code, this);
        editText = findViewById(R.id.input);
        textView = findViewById(R.id.text);
        textView.setOnClickListener(v -> {
            textView.setVisibility(GONE);
            editText.setVisibility(VISIBLE);
            editText.requestFocus();
            ScreenUtil.showSoftBoard(getContext(),editText);
        });
        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                if (editText.getText().length() == 6) {
                    if (listener != null)
                        listener.search(editText.getText().toString());
                }
            }
        });
    }

    public void setData(String code, String name) {
        textView.setVisibility(VISIBLE);
        editText.setVisibility(GONE);
        textView.setText(code + "  " + name);
        editText.setText("");
        ScreenUtil.hideSoftBoard(getContext(),editText);
    }

    OnSearchListener listener;

    public void setOnSearchListener(OnSearchListener listener) {
        this.listener = listener;
    }

    public interface OnSearchListener {
        void search(String code);
    }

    public EditText getEditText() {
        return editText;
    }
}

File: app/src/main/java/com/god/kotlin/trade/InputCodeView.kt
package com.god.kotlin.trade

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.util.AttributeSet
import android.view.View
import android.widget.EditText
import android.widget.RelativeLayout
import android.widget.TextView
import com.god.kotlin.R
import com.god.kotlin.util.hideSoftBoard
import com.god.kotlin.util.showSoftBoard

class InputCodeView : RelativeLayout {

    private var editText: EditText
    private var textView: TextView

    var listener: (String) -> Unit = {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        View.inflate(context, R.layout.trade_view_input_code, this)
        editText = findViewById(R.id.input)
        textView = findViewById(R.id.text)
        textView.setOnClickListener {
            textView.visibility = View.GONE
            editText.visibility = View.VISIBLE
            editText.requestFocus()
            editText.showSoftBoard(getContext())
        }
        editText.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

            override fun afterTextChanged(s: Editable) {
                if (editText.text.length == 6) {
                    listener(editText.text.toString())
                }
            }
        })
    }

    fun setData(code: String, name: String) {
        textView.visibility = View.VISIBLE
        editText.visibility = View.GONE
        textView.text = "$code  $name"
        editText.setText("")
        editText.hideSoftBoard(context)
    }

    fun setOnSearchListener(listener: (String) -> Unit) {
        this.listener = listener
    }
}
